toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -16,29 +16,29 @@ import (
 
 // SystemConfig defines how each system present on the image is supposed to be configured.
 type SystemConfig struct {
-	IsDefault            bool               `json:"IsDefault"`
-	IsKickStartBoot      bool               `json:"IsKickStartBoot"`
-	IsIsoInstall         bool               `json:"IsIsoInstall"`
-	BootType             string             `json:"BootType"`
-	Hostname             string             `json:"Hostname"`
-	Name                 string             `json:"Name"`
-	PackageLists         []string           `json:"PackageLists"`
-	Packages             []string           `json:"Packages"`
-	KernelOptions        map[string]string  `json:"KernelOptions"`
-	KernelCommandLine    KernelCommandLine  `json:"KernelCommandLine"`
-	AdditionalFiles      map[string]string  `json:"AdditionalFiles"`
-	PartitionSettings    []PartitionSetting `json:"PartitionSettings"`
-	PreInstallScripts    []InstallScript    `json:"PreInstallScripts"`
-	PostInstallScripts   []InstallScript    `json:"PostInstallScripts"`
-	FinalizeImageScripts []InstallScript    `json:"FinalizeImageScripts"`
-	Networks             []Network          `json:"Networks"`
-	PackageRepos         []PackageRepo      `json:"PackageRepos"`
-	Groups               []Group            `json:"Groups"`
-	Users                []User             `json:"Users"`
-	Encryption           RootEncryption     `json:"Encryption"`
-	RemoveRpmDb          bool               `json:"RemoveRpmDb"`
-	ReadOnlyVerityRoot   ReadOnlyVerityRoot `json:"ReadOnlyVerityRoot"`
-	HidepidDisabled      bool               `json:"HidepidDisabled"`
+	IsDefault            bool                      `json:"IsDefault"`
+	IsKickStartBoot      bool                      `json:"IsKickStartBoot"`
+	IsIsoInstall         bool                      `json:"IsIsoInstall"`
+	BootType             string                    `json:"BootType"`
+	Hostname             string                    `json:"Hostname"`
+	Name                 string                    `json:"Name"`
+	PackageLists         []string                  `json:"PackageLists"`
+	Packages             []string                  `json:"Packages"`
+	KernelOptions        map[string]string         `json:"KernelOptions"`
+	KernelCommandLine    KernelCommandLine         `json:"KernelCommandLine"`
+	AdditionalFiles      map[string]FileConfigList `json:"AdditionalFiles"`
```

Comment:
Not sure which I prefer here.. `map[string]FileConfigList` or `map[string][]FileConfig`

Reply:
`FileConfigList` only exists to allow me to attach a custom marshaller to it.